public/app/features/datasources/state/actions.ts

```diff
@@ -123,6 +125,50 @@ export function loadDataSource(id: number): ThunkResult<void> {
   };
 }
 
+/**
+ * Get data source by uid or id, if old id detected handles redirect
+ */
+async function getDataSourceUsingUidOrId(uid: string): Promise<DataSourceSettings> {
```

Comment:
Wouldn't it be easier to allow both string and number here and then make API call according to the type? Or is it not always given that `id` will be a number and `uid` a string?